### ISSUE TITLE
Weird lag with doChunkMap

### ISSUE THREAD

**Issue by iNko97** [lishid/Orebfuscator#177]
There is something weird happening with packets and my players on my server, that is impacting TPS.

Aikar's timings tell me this: (this problem disappears when Orebfuscator is disabled).

Minecraft::world - doChunkMap - Pending Send To Playerscount(17)  total(40.83% 1.643s, 74.67% of tick)avg(96.64ms per - 37.34ms/0.39 per tick

[Timings](https://timings.aikar.co/?id=413f373350de4895b350fa81579f3479) where this was extracted from:
Running latest PaperSpigot build, latest ProtocolLib and latest Orebfuscator.

How do the players notice this and how is this a problem? Other than impacting the tps a little, if they run mapping mods client side: maps start to look like [this](https://i.imgur.com/Sksp1qj.jpg) after exploration. 
So basically what happens is that a top chunk in the top left resets to nothing, following a specific pattern all around the map (Like [so](https://i.imgur.com/vhPypl6.jpg): red box is an estimate of the view distance (10 chunks) and red circle is the chunk that the server tries to send the client, that for some reason gets stuck)

I have no idea how to troubleshoot this further, but let me know how I can be of any help.

**Comment by iNko97** [comment]
This is the list of plugins currently on the server:

Plugins (54): ArthropodEgg, Bastion, BottledExperience, Brewery, CastleGates, ChestCommands, Citadel, CivChat2, CivModCore, CombatTagPlus, CrazyCrates, CropControl, dynmap, EasyHelpOP, Essentials, ExilePearl, FactoryMod, FastAsyncWorldEdit, Finale, HiddenOre, HiddenOre-Compat, Humbug, Imperivm, ItemExchange, JukeAlert, LibsDisguises, Multiverse-Core, MyCommand, MythicMobs, NameLayer, NoCheatPlus, OldEnchant, OpenTerrainGenerator, Orebfuscator4, PacketWrapper, ParkerCore, PlugMan, PortalFix, ProtocolLib, RandomSpawn, RealisticBiomes, RecipeBookDupePatch, RecipeManager, SimpleAdminHacks, TitleManager, UltraCosmetics, VanishNoPacket, ViaBackwards, ViaVersion, VipLogin, WordBank, WorldBorder, WorldEdit, XPSpawners

I wonder if the cause is Hiddenore-compat, since it calls some method in orebfuscator that is used solely by it and castlegates (Makes sense since this is quite a used plugin for nobody to notice this).

**Comment by iNko97** [comment]
The plugins I found are using packet magic are these:

ProtocolLib (4.3.0)
LibsDisguises (9.5.1-SNAPSHOT), (Thus UltraCosmetics, MythicMobs)
VanishNoPacket (3.2),
SimpleAdminHacks (1.2.3)

ViaBackwards (2.3.1), 
ViaVersion (1.3.0)

Plugins using Orebfuscator as a dependency:
HiddenOre-compat (0.0.1)
CastleGates (1.0.22-SNAPSHOT)

Does any of these ring a bell?

**Comment by ProgrammerDan** [comment]
:O
